.github/workflows/build_ci.yml

```
@@ -0,0 +1,229 @@
+          os: linux
+          arch: aarch64
+          runner: ubuntu-20.04
+        - target: x86_64-unknown-linux-musl
+          os: linux
+          arch: x86_64
+          runner: ubuntu-20.04
+        - target: x86_64-pc-windows-msvc
+          os: windows
+          arch: x86_64
+          runner: windows-2022
+          file_extension: '.exe'
+        - target: aarch64-pc-windows-msvc
+          os: windows
+          arch: aarch64
+          runner: windows-2022
+          file_extension: '.exe'
+        - target: x86_64-apple-darwin
+          os: macos
+          arch: x86_64
+          runner: macos-11
+        - target: aarch64-apple-darwin
+          os: macos
+          arch: aarch64
+          runner: macos-11
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Install rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        profile: minimal
+        override: true
+        target: ${{ matrix.target }}
+
+    - name: Install cargo-edit
+      uses: actions-rs/cargo@v1
+      with:
+        command: install
+        args: cargo-edit
+
+    - name: Set version
+      uses: actions-rs/cargo@v1
+      with:
+        command: set-version
+        args: ${{ env.VERSION }}
+
+    - name: Build target
+      uses: actions-rs/cargo@v1
+      with:
+        use-cross: true
+        command: build
+        args: --release --target ${{ matrix.target }}
+
+    - name: Zip
+      if: ${{ matrix.os == 'windows' }}
+      shell: pwsh
+      run: Compress-Archive "target/${{ matrix.target }}/release/lenra${{ matrix.file_extension }}" "lenra-${{ matrix.os }}-${{ matrix.arch }}.zip"
+    - name: Zip
+      if: ${{ matrix.os != 'windows' }}
+      shell: bash
+      run: tar -czf "lenra-${{ matrix.os }}-${{ matrix.arch }}.tar.gz" "target/${{ matrix.target }}/release/lenra${{ matrix.file_extension }}"
+
+    - name: Upload
+      uses: actions/upload-artifact@v3
+      with:
+        name: lenra-${{ matrix.os }}-${{ matrix.arch }}
+        path: lenra-${{ matrix.os }}-${{ matrix.arch }}.*
+
+
+  publish:
+    name: publish
+    needs: [build, get-next-version]
+    if: github.ref_name == 'main' || github.ref_name == 'beta'
+    runs-on: ubuntu-latest
+    env:
+      VERSION: ${{ needs.get-next-version.outputs.new-release-version }}
+    timeout-minutes: 8
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: download-artifacts
+      uses: actions/download-artifact@v3
+      with:
+        path: artifacts/
+
+    - name: Setup Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: "16"
+
+    - name: Setup node deps
+      shell: bash
+      run: npm i conventional-changelog-conventionalcommits@4 @semantic-release/git semantic-release-export-data -D
+
+    - id: release
+      name: Release
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: npx semantic-release@18
```

Maybe use it with only one line : 
```suggestion
      run: npx -p conventional-changelog-conventionalcommits@5 -p @semantic-release/git -p semantic-release-export-data -p semantic-release@19 semantic-release
```